langlib/lang.runtime/src/main/ballerina/runtime_errors.bal

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Type for value that can be cloned.
+# This is the same as in lang.value, but is copied here to avoid a dependency.
+type Cloneable readonly|xml|Cloneable[]|map<Cloneable>|table<map<Cloneable>>;
+
+# Error Detail Record
+public type Detail record {|
+   Cloneable...;
+|};
+
+// TypeCastError
+public type TypeCastError distinct error;
```

Comment:
This is a spec deviation. This is not in the https://ballerina.io/ballerina-spec/spec.html#lang.runtime. @manuranga

Reply:
Shall I move this `bal` file to `lang.error` module?

Reply:
@gabilang Hasitha is not suggesting to move it to a different module, he is just suggesting that we need to send a spec PR/issue first. Let's send that after we figure out the above cyclic dependency issue.